### PR TITLE
locally fetching data from db

### PR DESCRIPTION
**Pull request by EthanHogan**
@MilesSobhani Check this out when you get a chance! thanks!
Front end is successfully fetching data and populating state with that data.

### CODE REVIEW

File: client/components/App.jsx [hunk 1]
```diff
@@ -10,21 +11,49 @@ class App extends React.Component {
...
+      product_name: '',
+      price: '',
+      condition: false,
+      rating: 0,
+      review_count: 0,
+      availability: 0,
+      sale_count: 0,
+      seller: '',
+      sellerRating_count: 0,
+      seller_feedback: '',
```
Comment (EthanHogan) [inline client/components/App.jsx]: re-organized state and changed some of the key names to match up with key names in db.

File: client/components/App.jsx [hunk 2]
```diff
@@ -33,12 +62,12 @@ class App extends React.Component {
         <Head
           name = {this.state.product_name}
           rating = {this.state.rating}
-          numRatings = {this.state.numRatings}
+          reviewCount = {this.state.review_count}
```
Comment (EthanHogan) [inline client/components/App.jsx]: update naming to match db naming

File: client/components/App.jsx [hunk 3]
```diff
@@ -33,12 +62,12 @@ class App extends React.Component {
...
         />
         <br></br>
         <Subhead
           condition = {this.state.condition}
-          sold = {this.state.numSold}
+          sold = {this.state.sale_count}
```
Comment (EthanHogan) [inline client/components/App.jsx]: update naming to match db naming

File: client/components/App.jsx [hunk 4]
```diff
@@ -54,8 +83,8 @@ class App extends React.Component {
         <br></br>
         <Seller
           seller = {this.state.seller}
-          numRatings = {this.state.sellerRatings}
-          feedback = {this.state.sellerFeedback}
+          sellerRatingCount = {this.state.sellerRating_count}
+          sellerFeedback = {this.state.seller_feedback}
```
Comment (EthanHogan) [inline client/components/App.jsx]: update naming to match db naming

File: client/components/Head.jsx [hunk 5]
```diff
@@ -1,22 +1,22 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-const Head = ({ name, rating, numRatings }) => (
+const Head = ({ name, rating, reviewCount }) => (
```
Comment (EthanHogan) [inline client/components/Head.jsx]: update naming to match db naming

Comment (EthanHogan) [inline client/components/Head.jsx]: update naming to match db naming

File: client/components/Head.jsx [hunk 6]
```diff
@@ -1,22 +1,22 @@
...
         Head
         <br></br>
-        {name}
+        product_name: {name}
         <br></br>
-        {rating}
+        rating: {rating}
         <br></br>
-        {numRatings}
+        review_count: {reviewCount}
```
Comment (EthanHogan) [inline client/components/Head.jsx]: added text labels so i could see what data was what when it renders to my page

File: client/components/Head.jsx [hunk 7]
```diff
@@ -1,22 +1,22 @@
...
     </div>
 );
 
 Head.propTypes = {
   name: PropTypes.string.isRequired,
   rating: PropTypes.number.isRequired,
-  numRatings: PropTypes.number.isRequired,
+  reviewCount: PropTypes.number.isRequired,
```
Comment (EthanHogan) [inline client/components/Head.jsx]: update naming to match db naming

File: client/components/Logistics.jsx [hunk 8]
```diff
@@ -5,7 +5,7 @@ const Logistics = ({ shipping }) => (
     <div className= "logistics">
         Logistics
         <br></br>
-        {shipping}
+        shipping: {shipping}
```
Comment (EthanHogan) [inline client/components/Logistics.jsx]: added text labels so i could see what data was what when it renders to my page

File: client/components/Price.jsx [hunk 9]
```diff
@@ -5,7 +5,7 @@ const Price = ({ price }) => (
     <div className= "price">
         Price
         <br></br>
-        {price}
+        price: {price}
```
Comment (EthanHogan) [inline client/components/Price.jsx]: added text labels so i could see what data was what when it renders to my page

File: db/db.js [hunk 10]
```diff
@@ -37,9 +37,9 @@ const Model = mongoose.model('product', productSchema);
 
 // Model.insertMany(data);
 
-function getAllData() {
+function getData(id) {
```
Comment (EthanHogan) [inline db/db.js]: change this query name to be match its new functionality of only grabbing some data and not ALL data like it was before refactoring.

File: db/db.js [hunk 11]
```diff
@@ -37,9 +37,9 @@ const Model = mongoose.model('product', productSchema);
...
   return new Promise((resolve, reject) => {
-    Model.find({}, (err, results) => {
+    Model.find({ id }, (err, results) => {
```
Comment (EthanHogan) [inline db/db.js]: now only querying the data at the given id instead of all of the data

File: db/db.js [hunk 12]
```diff
@@ -49,4 +49,4 @@ function getAllData() {
   });
 }
 
-module.exports = { getAllData };
+module.exports = { getData };
```
Comment (EthanHogan) [inline db/db.js]: change this query name to be match its new functionality of only grabbing some data and not ALL data like it was before refactoring.

**Comment by MilesSobhani** [comment]
Looks fine!

**Pull request merged**